### ISSUE TITLE
Improve error transparency when tools are not installed in the configured venv

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#89]
## Summary

When pytest, pylint, or mypy is not installed in the configured Python environment, the MCP tool surfaces misleading generic errors instead of the real cause. The user has to dig through log files to find messages like `No module named pytest`. This issue improves error transparency across all three tools consistently.

## Implementation Plan

Based on discussion, the following approach was agreed:

### 1. Startup validation with cached state (all three tools)

On server startup, validate that `python -m pytest --version`, `python -m pylint --version`, and `python -m mypy --version` succeed using the **resolved** python executable (after venv path resolution, same logic as in `run_tests`).

- **Behavior**: Store results in `self._tool_availability: dict[str, bool]` on the `CodeCheckerServer` instance. Log a clear warning per missing tool, but **allow the server to start**.
- **Warning message includes guidance**: e.g., *"pytest not found in configured python-executable (/path/to/python.exe). Ensure --python-executable and --venv-path point to the tool's own environment where pytest is installed, not the project's runtime venv."*
- **At call time**: If a tool was flagged as missing at startup, **immediately return a clear error** without running the subprocess. This is a short-circuit -- no re-check at call time.
- **Trade-off**: If the user installs the tool after server start, they must restart the server. This is acceptable since environment changes typically require a restart anyway.

**Files**: `server.py` (add validation in `__init__`, add availability check in each tool handler)

### 2. Surface stderr on failure (truncated to 500 chars, all three tools)

When a tool subprocess exits with a non-zero code and produces no usable output (no JSON report for pytest, no parseable output for pylint/mypy), include the stderr output (truncated to 500 characters) in the error returned to the user.

**Pytest -- current code** (`runners.py`):
```python
raise RuntimeError(
    "Test execution completed but no report file was generated. "
    "Check for configuration errors in pytest.ini or pytest plugins."
)
```
**Fix**: Append `combined_output` (already computed at that point) truncated to 500 chars.

**Pylint/Mypy**: Similar -- when `execution_error` is returned, ensure stderr content is included in the error message if not already present.

**Truncation**: Use a shared constant `MAX_STDERR_IN_ERROR = 500` (e.g., in a shared utils module).

**Files**: `code_checker_pytest/runners.py`, `code_checker_pylint/runners.py`, `code_checker_mypy/runners.py`

### 3. Distinguish "tool missing" from "tool run failed" (all three tools)

Detect `"No module named pytest"` (and `pylint`, `mypy` equivalents) in stderr and raise a specific, actionable error message instead of the generic one.

- **Detection pattern**: Simple string match on `"No module named <tool>"` -- covers all platforms
- **No need for Windows-specific patterns** like `"is not recognized"` -- the venv resolution code already raises `FileNotFoundError` if the executable doesn't exist
- **Fallback**: If the pattern doesn't match, the stderr surfacing from part 2 ensures users still see the real error

**Files**: `code_checker_pytest/runners.py`, `code_checker_pylint/runners.py`, `code_checker_mypy/runners.py`

### 4. Remove `collect_environment_info()`

`collect_environment_info()` in `utils.py` makes 4+ subprocess calls on every pytest run (pytest version, pip list, trace-config, CPU/memory info) using `sys.executable` instead of the configured executable. However, **the collected data is never surfaced to the MCP client** -- `_format_pytest_result_with_details()` in `server.py` does not use it.

**Decision**: Remove `collect_environment_info()` entirely rather than fixing it. It is dead weight adding latency to every test run. If environment info is needed in the future, it can be re-added as a separate feature with proper design.

- Remove `collect_environment_info()` from `utils.py`
- Remove the call site in `runners.py`
- Remove `environment_context` from `PytestReport` model (or make it always `None`)
- Remove `environment_info` from the result dict in `check_code_with_pytest`
- Clean up `EnvironmentContext` model if no longer used

**Files**: `code_checker_pytest/utils.py`, `code_checker_pytest/runners.py`, `code_checker_pytest/models.py`

### 5. Documentation: README + CLI help text

- **README**: Add a "Configuration" / "Troubleshooting" section clarifying that `--python-executable` / `--venv-path` should point to the **tool's own venv** (where pytest, pylint, mypy are installed), not the project's runtime venv. Include correct vs. incorrect `.mcp.json` examples.
- **CLI help**: Improve `--help` descriptions for `--python-executable` and `--venv-path` to include this guidance.

**Files**: `README.md`, `main.py` (argparse help strings)

## Acceptance Criteria

- [ ] Server startup checks all three tools via `_tool_availability` dict and logs clear warnings with guidance if any are missing
- [ ] When a missing tool's endpoint is called, an immediate clear error is returned (short-circuit, no subprocess)
- [ ] When any tool fails with "No module named <tool>", the error explicitly says the tool is not installed and suggests checking the configured environment
- [ ] When any tool fails for other reasons with no usable output, stderr (up to 500 chars) is included in the error
- [ ] `collect_environment_info()` and related dead code are removed
- [ ] CLI `--help` for `--python-executable` and `--venv-path` mentions they should point to the tool's venv
- [ ] README documents correct environment setup with examples and troubleshooting
- [ ] Error handling is consistent across all three tools (pytest, pylint, mypy)
- [ ] All existing tests pass; new tests cover the validation and error scenarios

## Impact

Users get immediate, actionable feedback instead of spending significant time searching log files for the real error.

---

# Original issue: Improve error transparency when pytest is not installed in the target venv

## Problem

When pytest is not installed in the configured project venv, the MCP tool surfaces a misleading generic error to the user:

> `Error running pytest: Test execution completed but no report file was generated. Check for configuration errors in pytest.ini or pytest plugins.`

The **actual** root cause is logged internally but never shown:

> `C:\...\mcp_server_filesystem_57\.venv\Scripts\python.exe: No module named pytest`

This makes debugging very hard -- the user has to dig through log files to find the real problem.

## Root cause: likely a misconfiguration in `.mcp.json`

The deeper issue is that `--python-executable` and `--venv-path` should point to the **same environment that runs the MCP code-checker tool itself** (`VIRTUAL_ENV`), not to the project's local venv (`MCP_CODER_VENV_DIR`).

The MCP code-checker tool is a dev tool -- it owns pytest, pylint, mypy etc. The project under test should not need to have these installed. A typical misconfiguration looks like:

```json
{
  "code-checker": {
    "command": "${VIRTUAL_ENV}\\Scripts\\mcp-code-checker.exe",
    "args": [
      "--python-executable", "${MCP_CODER_VENV_DIR}\\Scripts\\python.exe",
      "--venv-path",         "${MCP_CODER_VENV_DIR}"
    ]
  }
}
```

The `--python-executable` and `--venv-path` point to `MCP_CODER_VENV_DIR` (the project venv), but pytest lives in `VIRTUAL_ENV` (the tool's venv). The correct configuration is:

```json
{
  "code-checker": {
    "command": "${VIRTUAL_ENV}\\Scripts\\mcp-code-checker.exe",
    "args": [
      "--python-executable", "${VIRTUAL_ENV}\\Scripts\\python.exe",
      "--venv-path",         "${VIRTUAL_ENV}"
    ]
  }
}
```

This is easy to get wrong because both variables look similar and the error message gives no hint about the environment mismatch.

## Reproduction

1. Configure `--python-executable` / `--venv-path` to point to a venv that does **not** have `pytest` installed (e.g. the project's runtime venv).
2. Call `run_pytest_check`.
3. The returned error message mentions "no report file was generated" and hints at pytest.ini -- not the real cause.

## What was observed in the logs

From the debug log, the subprocess runner logs:
```
Test collection error occurred (code 1), but continuing execution:
C:\..\.venv\Scripts\python.exe: No module named pytest
```

...but then raises `RuntimeError("Test execution completed but no report file was generated.")` -- discarding the real stderr.

There is also a secondary issue: the tool first probes using the **host** venv with a 10-second timeout, which times out before falling through to the configured venv. This probe result is also not surfaced to the user.

## Proposed improvements

1. **Validate environment parameters on startup**: On server start, check that `--python-executable` can run `python -m pytest --version`. If it fails, log a clear warning:
   > `pytest not found in configured python-executable. Check that --python-executable and --venv-path point to the tool's own environment (VIRTUAL_ENV), not the project's runtime venv.`

2. **Surface stderr on failure**: When the pytest subprocess exits with a non-zero code and produces no JSON report, include the raw stderr in the error returned to the user -- not just a generic hint about pytest.ini.

3. **Distinguish "pytest missing" from "test run failed"**: The two situations currently produce the same error. They should be differentiated so the user knows whether it is a configuration problem or a test failure.

4. **Document the intended environment setup**: Make clear in the README that `--python-executable` / `--venv-path` should point to the **tool's own venv** (where pytest, pylint, mypy are installed), while `--project-dir` and `PYTHONPATH` cover the project under test.

## Impact

Users spend significant time searching log files for the real error, which defeats the purpose of a developer-friendly MCP tool.
